packages/actor-query-operation-filter-direct/lib/ActorQueryOperationFilterDirect.ts

```diff
@@ -16,11 +19,20 @@ export class ActorQueryOperationFilterDirect extends ActorQueryOperationTypedMed
   public async testOperation(pattern: Algebra.Filter, context: ActionContext): Promise<IActorTest> {
     // will throw error for unsupported operators
     SparqlExpressionEvaluator.createEvaluator(pattern.expression);
+
+    // Check if the source is a hypermedia tree, because this filter can't handle that
+    const source = await ActorQueryOperationSparqlEndpoint.getSingleSource(context);
+    if (source && source.type === "hypermedia-tree") {
```

Comment:
I'm actually not convinced that this actor does not work on sources of the type hypermedia tree. It can still pull in all triples from that source and then apply the filters locally. And only for a few select filters would this be a difference. (I would suggest looking into applying a mediatortype to return a different value in the test function for these specific instances)